provider/datagen/src/bin/datagen.rs

```diff
@@ -135,12 +134,11 @@ fn main() -> eyre::Result<()> {
                 ),
         )
         .arg(
-            Arg::with_name("KEY_FILE")
-                .long("key-file")
+            Arg::with_name("BINARY")
+                .long("binary")
```

Comment:
Nit: I'm still not completely onboard with this much indirection and removing the key file. But, if you include this option, please name it something such as:

- `--key-binary`
- `--inspect`

Reply:
Also, I don't really think that key-extract will be everyone's preferred workflow: i still think supporting key files is useful.

Reply:
that seems ok to me